### PR TITLE
Add unzip instance for Getter

### PR DESCRIPTION
**Pull request by lunaryorn**
Closes GH-455

@julien-truffaut While I was at it :)

### CODE REVIEW

**Comment by julien-truffaut** [comment]
if you get the time, could you also add `unzip` to the `Getter` class, the signature should probably look like this:

```
import scalaz.Leibniz.===
def unzip[B, C](implicit ev: (B, C) === A): (Getter[S, B], Getter[S, C])
```

but we should check that type inference works with something like:
```
getterA.zip(getterB).unzip
```

see also https://github.com/julien-truffaut/Monocle/blob/master/core/shared/src/main/scala/monocle/Prism.scala#L246

**Comment by lunaryorn** [comment]
I added the instance method to `Getter`, but it doesn't look like type inference can cope with it.  The compiler won't find the implicit evidence for `===` unless you explicitly give type parameters to `unzip`.

File: test/shared/src/test/scala/monocle/GetterSpec.scala [hunk 1]
```diff
@@ -57,4 +64,26 @@ class GetterSpec extends MonocleSuite {
...
+    length.get("helloworld") shouldEqual 10
+    upper.get("helloworld") shouldEqual "HELLOWORLD"
+
+  }
+
+  test("zip/unzip roundtrip") {
+    val length = Getter[String, Int](_.length)
+    val upper = Getter[String, String](_.toUpperCase)
+
+    val (length1, upper1) = length.zip(upper).unzip[Int, String]
```
Comment (lunaryorn) [inline test/shared/src/test/scala/monocle/GetterSpec.scala]: I had to give the type parameters to `.unzip` explicitly to make the compiler find the implicit evidence for `===`.  Without explicit type parameters it wouldn't compile.

Comment (julien-truffaut) [inline test/shared/src/test/scala/monocle/GetterSpec.scala]: arf ok, maybe we should revert back to your first commit if we can't get the type inference to work properly

**Comment by lunaryorn** [comment]
@julien-truffaut I dropped the `.unzip` instance method again.

**Comment by julien-truffaut** [comment]
@lunaryorn sorry for the back and force. Thank you very much 👍

**Comment by lunaryorn** [comment]
@julien-truffaut Never mind; I learned a bit about scalaz along the way for I never actually used it before.

**Pull request merged**